src/mqt/yaqs/digital/digital_tjm.py

```
@@ -522,44 +529,57 @@ def digital_tjm(
+    """Return this trajectory's share of the total ``shots`` budget.
+
+    For combined noisy runs, ``WORKER_CTX["shot_distribution"]`` holds
+    ``(total_shots, num_traj)`` and this may return ``0`` for some ``traj_idx``.
 
     Returns:
-        Number of shots for the current worker invocation.
+        Non-negative shot count for the current worker invocation.
     """
-    try:
-        from mqt.yaqs.simulator import WORKER_CTX  # ruff:ignore[import-outside-top-level]
+    from mqt.yaqs.simulator import WORKER_CTX  # ruff:ignore[import-outside-top-level]
```

Ruff (0.16.0)

[warning] Invalid rule code in suppression: import-outside-top-level

Enable `lint.preview` to use rule names

(RUF102)